### DISCUSSION TITLE
Why rectangular boxes cannot be drawn and displayed on the background or sky box

### DISCUSSION THREAD

**Discussion by Haidiye00** [google/filament#7032] [Q&A]
this is my shader and the vertex submission is specified as an NDC coordinate
![zYoMJkpU6l](https://github.com/google/filament/assets/132528822/3ededa2b-54af-446d-9f8d-a7514777eb16)
This is a normal drawing
![0Nl0bvufd0](https://github.com/google/filament/assets/132528822/0b311023-3857-42dc-bab8-78876cfeae4e)
![aqPNY1czRq](https://github.com/google/filament/assets/132528822/8d312c1e-d040-4a46-bf75-0cff69e8fb64)
and I change the data of vertices，I found that the area of the sky box cannot draw a rectangular box，May I ask how to solve it？
![Q96LDHwyHc](https://github.com/google/filament/assets/132528822/93e76ca8-eb3c-4227-915f-abc3380157e1)
![Uwy55uc4mZ](https://github.com/google/filament/assets/132528822/26e57664-d711-48ab-80e7-16ab33b88a24)

**Comment by romainguy** [comment]
Please don't use screenshots to show code. Anyway it could be due to ordering of the draw calls since the skybox is set to [draw last by default](https://github.com/google/filament/blob/51c65ccfdcd39546503bc3323b19d702c509dcdd/filament/src/details/Skybox.cpp#L114). You could also solve it by using `depthWrite` and giving your rectangle a depth of 0.9999f instead of 1f so the skybox always draws behind it.

**Comment by Haidiye00** [comment]
Our company has information security and cannot copy code, you understand. Thank you for your guidance. I'll give it a try. Thank you very much~
